### PR TITLE
fix!: Fix Host Allow List Handler

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
fix!: Fix Host Allow List Handler
 --handle case of empty Host header and its value

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

## Testing

- [x] I thought about testing these changes in a realistic/non-local environment.

### CODE REVIEW

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/HostAllowListHandler.java [hunk 1]
```diff
@@ -51,15 +48,8 @@ public void handle(final RoutingContext event) {
...
-    final long hostPieces = stream(splitHostHeader).count();
-    if (hostPieces > 1) {
-      // If the host contains a colon, verify the host is correctly formed - host [ ":" port ]
-      if (hostPieces > 2 || !Iterables.get(splitHostHeader, 1).matches("\\d{1,5}+")) {
-        return Optional.empty();
-      }
-    }
-    return Optional.ofNullable(Iterables.get(splitHostHeader, 0));
+    final HostAndPort hostAndPort = event.request().authority();
+    return Optional.ofNullable(hostAndPort).map(HostAndPort::host);
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/HostAllowListHandler.java:52]: This is a nice improvement

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/HttpHostAllowListAcceptanceTest.java [hunk 2]
```diff
@@ -80,4 +92,39 @@
...
+            .withHttpAllowHostList(Collections.singletonList("127.0.0.1"))
+            .withMode("eth2")
+            .build();
+    startSigner(signerConfiguration);
+
+    // raw request without Host header
+    final URI uri = URI.create(signer.getUrl());
+    try (Socket s = new Socket(InetAddress.getLoopbackAddress(), uri.getPort())) {
+      final PrintWriter out =
+          new PrintWriter(new OutputStreamWriter(s.getOutputStream(), UTF_8), true);
```
[the remark this answers is not in the document]
Comment (jframe) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/HttpHostAllowListAcceptanceTest.java:111]: Can the PrinterWriter be moved into the try with then this resource leak warning should go away.

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/HttpHostAllowListAcceptanceTest.java [hunk 3]
```diff
@@ -80,4 +92,39 @@
...
+              + UPCHECK_ENDPOINT
+              + " HTTP/1.1\r\n"
+              + "Connection: close\r\n" // signals server to close the connection
+              + rawHeader
+              + "\r\n"; // end of headers section
+      out.write(req);
+      out.flush();
+
+      final BufferedReader br =
+          new BufferedReader(new InputStreamReader(s.getInputStream(), UTF_8));
```
[the remark this answers is not in the document]
Comment (jframe) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/HttpHostAllowListAcceptanceTest.java:123]: Similar here can this block with the BufferedReader be done in a try with to take care of the resource leak

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/HttpHostAllowListAcceptanceTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (usmansaleem) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/HttpHostAllowListAcceptanceTest.java]: done

**Review by jframe — APPROVED** [review]

**Pull request merged**